### DISCUSSION TITLE
[help] routing like nextjs docs

### DISCUSSION THREAD

**Discussion by scr2em** [vercel/next.js#17862] [Help] [answered]
hello there, I'm new to NextJs so no need to flame me.
![image](https://user-images.githubusercontent.com/4671486/95939115-a9a39100-0ddb-11eb-95cc-1d40a00fe7f0.png)
how do the fetch the right side (with url change as well) without having to load the other components again ?

I'm trying to build **/profile** with **/profile/settings** and **/profile/oders** 
my left side (profile picture and some user data),navbar and footer are the same for the 3 routes while my right side the only one changing 
currently im using the whole Profile component for for **/profile/settings** and **/profile/orders** as well (with some little changes to the right side)  
folder structure 
--profile 
    ->index.js
    ->settings.js
    ->orders.js

how do I load a certain part of the page with url change and the user dont get a 404 when opening the page of the first time 
**just like nextjs docs** ?

**Comment by jamesmosier** [comment, marked as answer]
You can put your sidebar navigation in a [custom app file](https://nextjs.org/docs/advanced-features/custom-app) and then use [next/link](https://nextjs.org/docs/api-reference/next/link) to navigate between pages.

Or as the Next docs does it, you can use [optional catch all routes](https://nextjs.org/docs/routing/dynamic-routes#optional-catch-all-routes)

For example:

`[[...profile]].js`
```js
import Link from 'next/link';

export default function Profile({ data }) {
  return (
    <div>
      <aside>
        <Link href="/profile/settings">
          <a>Settings</a>
        </Link>
        <br />
        <Link href="/profile/orders">
          <a>Orders</a>
        </Link>
      </aside>
      <div>This is a page with data: {data}</div>
    </div>
  );
}

export async function getStaticProps(context) {
  return {
    props: { data: JSON.stringify(context.params) },
  };
}

export async function getStaticPaths() {
  return {
    paths: [{ params: { profile: ['settings', 'orders'] } }],
    fallback: true,
  };
}
```

**Reply by scr2em** [reply]
thanks, this really helped me.
also what if i want to check authentication server side ? because i cant use getServerSideProps to check the incoming req

**Reply by jamesmosier** [reply]
The only way to check authentication server side would be `getServerSideProps`. 

Could you just check authentication on the client side? For example via an API call to check the users authentication status?

**Reply by scr2em** [reply]
thanks for the fast reply. Ill try that
